source/postprocess/boundary_strain_rate_residual_statistics.cc

```diff
@@ -0,0 +1,333 @@
+                                       + " (" + units + ")";
+          statistics.add_value (name_rms, rms->second*unit_scale_factor);
+          // also make sure that the other columns filled by this object
+          // all show up with sufficient accuracy and in scientific notation
+          statistics.set_precision (name_max, 8);
+          statistics.set_scientific (name_max, true);
+          statistics.set_precision (name_min, 8);
+          statistics.set_scientific (name_min, true);
+          statistics.set_precision (name_rms, 8);
+          statistics.set_scientific (name_rms, true);
+
+          screen_text << max_eii->second *unit_scale_factor << " " << units << ", "
+                      << min_eii->second *unit_scale_factor << " " << units << ", "
+                      << rms->second *unit_scale_factor << " " << units
+                      << (index == global_max_eii.size()-1 ? "" : ", ");
+
+        }
+
+      return std::pair<std::string, std::string> ("Max, min, and RMS residual strain rate invariant along boundary parts:",
+                                                  screen_text.str());
+    }
+
+
+
+    template <int dim>
+    void
+    BoundaryStrainRateResidualStatistics<dim>::declare_parameters (ParameterHandler  &prm)
+    {
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Boundary strain rate residual statistics");
+        {
+          prm.declare_entry ("Data directory",
+                             "$ASPECT_SOURCE_DIR/data/postprocess/boundary-strain-rate-residual/",
+                             Patterns::DirectoryName (),
+                             "The name of a directory that contains the ascii data. "
+                             "This path may either be absolute (if starting with a "
+                             "`/') or relative to the current directory. The path may also "
+                             "include the special text `$ASPECT_SOURCE_DIR' which will be "
+                             "interpreted as the path in which the ASPECT source files were "
+                             "located when ASPECT was compiled. This interpretation allows, "
+                             "for example, to reference files located in the `data/' subdirectory "
+                             "of ASPECT.");
+          prm.declare_entry ("Data file name", "box_3d_boundary_strain_rate.txt",
+                             Patterns::Anything (),
+                             "The file name of the input surface strain rate an ascii data. "
+                             "The file has (dim-1)*(dim-1) columns in addition to the coordinate columns. "
+                             "The strain rate data columns correspond to the different strain rate components "
+                             "at the surface, i.e. 2 components for 2d models and 4 components for 3d models. ");
```

Comment:
Is this true or is it only one column with the strain rate invariant?

Reply:
This is not true anymore. Thanks for catching it!